script/mail_fetcher

```diff
@@ -38,8 +38,11 @@ handler = FactCheckEmailHandler.new(Publisher::Application.fact_check_config)
 AUTOMATIC_LOCK_EXPIRY = (5 * 60) # seconds
 begin
   Redis.new.lock("publisher:#{Rails.env}:fact_check_processing_lock", life: AUTOMATIC_LOCK_EXPIRY) do |lock|
-    handler.process do
-      lock.extend_life(AUTOMATIC_LOCK_EXPIRY)
+    Redis.new.lock("publisher:fact_check_processing_lock", life: AUTOMATIC_LOCK_EXPIRY) do |inner_lock|
+      handler.process do
+        inner_lock.exception(AUTOMATIC_LOCK_EXPIRY)
+        lock.extend_life(AUTOMATIC_LOCK_EXPIRY)
```

Comment:
Should this be `extend_life` rather than `exception`?

```suggestion
        inner_lock.extend_life(AUTOMATIC_LOCK_EXPIRY)
        lock.extend_life(AUTOMATIC_LOCK_EXPIRY)
```

Reply:
Yep, it should be. I don't know how I missed that!